src/hooks/useRelatedVenues.tsx

```diff
@@ -23,7 +28,7 @@ export interface RelatedVenuesContextState {
   relatedVenueIds: string[];
 
   findVenueInRelatedVenues: (
-    searchedForVenueId?: string
+    searchOptions: FindVenueInRelatedVenuesArgs
```

Comment:
It's clear it's search `options`, just like `i` in `for`. 

The interesting thing, though, since all options are... khm... optional, the function shouldn't compel invocations to provide empty object e..g. `find()` over `find({})`.

Bar the above, if there is some kind of exclusion principle going on... why not 2 different functions: `findById`, `findBySlug`?

```suggestion
    options?: FindVenueInRelatedVenuesArgs
```

Reply:
This whole Provider already has far too much going on in my view. Rather than adding yet another function to it I overloaded the existing. Maybe we revisit that when we get rid of all the references to sovereign venue.